web/libs/editor/src/stores/Annotation/Annotation.js

```diff
@@ -1222,21 +1219,24 @@ const _Annotation = types
 
         self._initialAnnotationObj = objAnnotation;
 
-        objAnnotation.forEach((obj) => {
+        for (const obj of objAnnotation) {
           self.deserializeSingleResult(
             obj,
             (id) => areas.get(id),
             (snapshot) => areas.put(snapshot),
           );
-        });
+        }
 
         // It's not necessary, but it's calmer with this
         self.cleanClassificationAreas();
 
-        !hidden &&
-          self.results.filter((r) => r.area.classification).forEach((r) => r.from_name.updateFromResult?.(r.mainValue));
+        if (!hidden) {
+          for (const r of self.results.filter((r) => r.area.classification)) {
+            r.from_name.updateFromResult?.(r.mainValue);
+          }
+        }
```

Comment:
if i'm not mistaken, filter will run on each iteration. it's safer and cleaner to move it to a separate variable

```suggestion
        if (!hidden) {
          const filteredResults = self.results.filter((r) => r.area.classification); 
          for (const r of filteredResults) {
            r.from_name.updateFromResult?.(r.mainValue);
          }
        }
```